src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1092,14 +1079,14 @@ static int volume_process(struct processing_module *mod,
 			frames = cd->vol_ramp_frames;
 		}
 
-		/* copy and scale volume */
-		cd->scale_vol(mod, &input_buffers[0], &output_buffers[0], frames);
+		if (!cd->ramp_finished)
```

Comment:
I like put it ahead of volume, and if the condition based on ramp_sample_left will be better.

Reply:
Yep, agree. I'd like to get rid of duplicate ramp_finished and vol_ramp_active but I'm not sure how to do it. So I'm keeping this still draft. Also I have problems with check_volume_levels.sh stability so I'm not 100% this in stress test always passes.

Reply:
you can create an jira ticket or github issue, then we can do it when we have free time: to simply those flags.